experiments_main.cpp

```diff
@@ -201,8 +214,62 @@ class ExperimentRunner {
         generate_sequences();
         std::cout << "Computing edit distances ... ";
         compute_edit_distance();
-        apply_tuple([&](auto &algorithm, auto &dist) { run_sketch_algorithm(&algorithm, &dist); },
-                    algorithms, dists);
+        apply_tuple(
+                [&](auto &algorithm, auto &dist) {
+                    std::cout << "Running " << algorithm.name << std::endl;
+
+                    // Run the algorithms FLAGS_reruns times, storing the distances and Spearman
+                    // coefficient computed in each run. The average and standard deviation of the
+                    // Spearman coefficients is reported, as well as the Spearman coefficient
+                    // obtained from using the median and average of the distances of all runs.
+                    vector<double> spearman_coefficients(FLAGS_reruns);
+                    Vec2D<double> dists_per_run = new2D<double>(FLAGS_reruns, ingroup_pairs.size());
+
+                    for (uint32_t rerun = 0; rerun < FLAGS_reruns; ++rerun) {
+                        spearman_coefficients[rerun]
+                                = run_sketch_algorithm(&algorithm, &dists_per_run[rerun]);
+                    }
+
+                    if (FLAGS_reruns > 1) {
+                        // Transpose of dists_per_run.
+                        Vec2D<double> runs_per_dist
+                                = new2D<double>(ingroup_pairs.size(), FLAGS_reruns);
+                        for (size_t i = 0; i < FLAGS_reruns; ++i)
+                            for (size_t j = 0; j < dists_per_run[i].size(); ++j)
+                                runs_per_dist[j][i] = dists_per_run[i][j];
+
+                        for (auto &distances : runs_per_dist)
```

Comment:
auto->double

For readability, the convention I used so far is to only use `auto` when the type is absolutely clear (e.g. with `make_unique`) or when the type is so complex that it reduces readability to write it (typically with iterators).

In this particular case, it saves teh code reader a lookup to figure out what type the distance actually is.

Reply:
I like `auto` personally, but I'll try not to  :)